internal/pkg/provision/providers/qemu/launch.go

```diff
@@ -0,0 +1,86 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package qemu
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"strconv"
+
+	"github.com/talos-systems/talos/internal/pkg/provision/providers/vm"
+)
+
+// LaunchConfig is passed in to the Launch function over stdin.
+type LaunchConfig struct {
+	DiskPath       string
+	VCPUCount      int64
+	MemSize        int64
+	QemuExecutable string
+}
+
+// Launch a control process around qemu VM manager.
+//
+// This function is invoked from 'talosctl qemu-launch' hidden command
+// and wraps starting, controlling 'qemu' VM process.
+//
+// Launch restarts VM forever until control process is stopped itself with a signal.
+//
+// Process is expected to receive configuration on stdin. Current working directory
+// should be cluster state directory, process output should be redirected to the
+// logfile in state directory.
+//
+// When signals SIGINT, SIGTERM are received, control process stops qemu and exits.
+//
+//nolint: gocyclo
+func Launch() error {
+	var config LaunchConfig
+
+	if err := vm.ReadConfig(&config); err != nil {
+		return err
+	}
+
+	c := vm.ConfigureSignals()
+
+	cmd := exec.Command(
+		config.QemuExecutable,
+		"-m", strconv.FormatInt(config.MemSize, 10),
+		"-drive", fmt.Sprintf("format=raw,file=%s", config.DiskPath),
+		"-smp", fmt.Sprintf("cpus=%d", config.VCPUCount),
+		"-accel",
+		"kvm",
+		"-nographic",
```

Comment:
probably we need `-kernel`, `-initrd` and `-append` for the first boot as disk image is empty

also not 100% sure how to do that, but we need to tell qemu to map in disk as virtio device (should be faster and it will have the same name `/dev/vda` as firecracker)

Reply:
looks like `,if=virtio` should be added to the `-drive` line